environment/config.go

```diff
@@ -16,14 +16,16 @@ type Configuration struct {
 	mu sync.RWMutex
 
 	environmentVariables []string
+	labels               map[string]string
 	settings             Settings
 }
 
 // Returns a new environment configuration with the given settings and environment variables
 // defined within it.
-func NewConfiguration(s Settings, envVars []string) *Configuration {
+func NewConfiguration(s Settings, envVars []string, labels map[string]string) *Configuration {
```

Comment:
Don't add an additional field here, merge it into the existing `Settings` struct.